circle.yml

```diff
@@ -137,6 +138,35 @@ test:
       do
         convert $dir/$x/*.png -append $dir/$x.png
       done
+      # Remove all directories in $CIRCLE_ARTIFACTS/img
+      # Currently the pieces aren't kept around. If it's
+      # desirable to keep them around, we should use
+      #    cp -r $dir $CIRCLE_ARTIFACTS/img_pieces
+      # The reason the image pieces aren't currently kept
+      # around is that it was leading to a problem and didn't
+      # want to sink too much time into the problem at the moment.
+      rm -R -- $dir/*/
+      # Install utility we need
+      npm install -g json
+      # Download the latest mathquill artifacts.
+      curl $(curl https://circleci.com/api/v1/project/mathquill/mathquill/latest/artifacts \
+              | json -a url pretty_path -d '\n\t' \
+              | grep '\.png$' \
+              | grep -v 'PREV' \
+              | sed "s:\$CIRCLE_ARTIFACTS/imgs/:-o $CIRCLE_ARTIFACTS/imgs/PREV_:")
+      cd $CIRCLE_ARTIFACTS/imgs/
+
+      for file in $(ls PINNED*); do
+        prev=PREV_$file
+        metric_diff=$(compare -metric AE -compose src $prev $file raw_diff.png)
+        composite -alpha on raw_diff.png $prev DIFF_$file
+      done
+
+      for file in $(ls EVERGREEN*); do
+        prev=$(ls PREV_$file*)
+        metric_diff=$(compare -metric AE -compose src $prev $file raw_diff.png)
+        composite -alpha on raw_diff.png $prev DIFF_$file
+      done
```

Comment:
I think it's preferable to separate the shell commands into separate `circle.yml` lines as much as possible, which is nice because CircleCI times them and displays output from them separately. I see they all share `$dir` which won't get shared across separate `circle.yml` lines, but you can easily add environment variables that all commands will see to `circle.yml`: https://circleci.com/docs/configuration/#environment